src/CodeViewer/Viewer.cpp

```diff
@@ -165,4 +165,33 @@ void Viewer::ClearHeatmapSource() {
   UpdateLeftSidebarWidth();
 }
 
+void Viewer::SetHighlightCurrentLine(bool enabled) {
+  if (highlight_current_line_ == enabled) return;
+
+  highlight_current_line_ = enabled;
+
+  if (!highlight_current_line_) {
+    QObject::disconnect(this, &Viewer::cursorPositionChanged, this, &Viewer::HighlightCurrentLine);
+    return;
+  }
+
+  QObject::connect(this, &Viewer::cursorPositionChanged, this, &Viewer::HighlightCurrentLine,
+                   Qt::UniqueConnection);
+  HighlightCurrentLine();
```

Comment:
I don't understand why this function is only called once and not every time that we select a different line, but as it is working, I don't have anything to complain.

Reply:
Well the `QObject::connect` ensure that it is called whenever the cursor position changes.

And then we just call it once to highlight the current line. Otherwise it would a cursor movement to start the highlighting.

Reply:
Nice!